File: 백준/Silver/27446. 랩실에서 잘 자요/랩실에서 잘 자요.py
import sys
input = sys.stdin.readline

n, m = map(int, input().split())
bd = list(map(int, input().split()))
pg = [i+1 for i in range(n)]
pr = [j for j in pg if j not in bd]

ink = 0
for k in range(len(pr)):
    if k == 0:
        ink += 7
    else:
        ink += min(7, 2*(pr[k] - pr[k-1]))

print(ink)

File: 백준/Silver/27446. 랩실에서 잘 자요/랩실에서 잘 자요.js
const input = require("fs")
  .readFileSync(process.platform === "linux" ? "/dev/stdin" : "./example.txt")
  .toString()
  .trim()
  .split("\n")
  .map((v) => v.split(" ").map(Number));

const [[n, m], pg] = input;

const page = [...new Array(n)].map((_, i) => i + 1);
const prt = page.filter((v) => !pg.includes(v));

let res = 0;
for (let i = 0; i < prt.length; i++) {
  if (i === 0) res += 7;
  else res += Math.min(7, 2 * (prt[i] - prt[i - 1]));
}
console.log(res);
